R/cat_add_data.R

```diff
@@ -7,15 +7,113 @@
+#' of each expert should be repeated as many times as there are impact levels
+#' and sites considered in the elicitation
+#' (i.e. each expert should appear \eqn{number\ of\ levels
+#' \cdot number\ of\ sites} times).
+#' * The second column should be the names of the levels considered in the
+#' eliciation. Each level should be repeated as many times as there are sites
+#' considered.
+#' * The third column should hold the names of the sites considered in the
+#' study. The name of each site should be repeated as many times as the number
+#' of levels.
+#' * The fourth column should be the experts confidence levels (given in
+#' percents). Experts should each give one confidence level per site, so each
+#' confidence level should be repeated for all levels at each site.
+#' * The final column should be the estimates of each expert.
+#'
+#' @section Data cleaning:
+#'
+#' When data are added to the [elic_cat] object, first names are standardised
+#' by converting capital letters to lower case, and by removing any whitespaces
+#' and punctuation. Then, data are anonymised by converting names to short sha1
+#' hashes. In this way, sensible information collected during the elicitation
+#' process never reaches the [elic_cat] object.
+#'
+#' If the data are imported from _Google Sheets_, `elic_cat_add_data()`
+#' performs additional data cleaning operations. This is relevant when data are
+#' collected with Google Forms because, for example, there could be multiple
+#' submission by the same expert or a different decimal separator could be used.
+#' When data are collected with Google Form, a column with the date and time is
+#' recorded. First, the function checks for multiple submissions and if present,
+#' only the last submission is retained. Second, the function removes the column
+#' with the timestamp. Then it checks for consistency of the decimal separator,
+#' i.e. commas _,_ are replaced with periods _._. Finally, columns four and five
+#' (which contain the confidence levels and estimates) are forced to numeric.
+#'
 #' @return The provided object of class [elic_cat] updated with the data.
 #' @export
 #'
 #' @family cat data helpers
 #'
 #' @examples
+#' # Create the elic_cat object for an elicitation process with three
+#' # mechanisms, four sites, five levels and a maximum of six experts per
+#' # mechanism
+#' x <- elic_cat_start(levels = c("levels_1", "levels_2", "levels_3",
+#'                                "levels_4", "levels_5"),
+#'                     sites = c("site_1", "site_2", "site_3", "site_4"),
+#'                     experts = 6,
+#'                     mechanisms = c("mechanism_1", "mechanism_2",
+#'                                    "mechanism_3"))
+#'
+#' # Add data for the three mechanisms from a data.frame. Notice that the
+#' # three commands can be piped
+#' my_elicit <- elic_cat_add_data(x, data_source = mechanism_1,
+#' mechanism = 1) |>
+#'   elic_cat_add_data(data_source = mechanism_2, mechanism = 2)
```

Comment:
Lines 73-75: here is missing the code for the third mechanism.

Moreover, the mechanism is provided as integer and not as character. Please, change it here and in the examples below. My point is that in this example it is quite clear which one is the first, second or third mechanism since we named them "mechanism_1", "mechanism_2" and "mechanism_3". But with real name this could be not so straightforward...